catalog/core/catalog-core-validator/src/main/java/ddf/catalog/validation/impl/ValidationParser.java

```diff
@@ -0,0 +1,258 @@
+            validators.put(entry.getKey(), lv);
+        }
+        outer.validators = validators;
+    }
+
+    private void parseAttributeTypes(Map<String, Outer.AttributeType> attributeTypes,
+            String registrationId) {
+        // TODO (RCZ) - Maybe do some sort of validation with these?
+        for (Map.Entry<String, Outer.AttributeType> e : attributeTypes.entrySet()) {
+            AttributeDescriptor descriptor = new AttributeDescriptorImpl(e.getKey(),
+                    e.getValue().indexed,
+                    e.getValue().stored,
+                    e.getValue().tokenized,
+                    e.getValue().multivalued,
+                    BasicTypes.getAttributeType(e.getValue().type));
+            attributeRegistry.registerAttribute(descriptor);
+
+        }
+    }
+
+    private void parseMetacardTypes(List<Outer.MetacardType> metacardTypes, String registrationId) {
+        BundleContext context = FrameworkUtil.getBundle(this.getClass())
+                .getBundleContext();
+        for (Outer.MetacardType metacardType : metacardTypes) {
+            Set<AttributeDescriptor> attributeDescriptors = new HashSet<>();
+            Set<String> requiredAttributes = new HashSet<>();
+
+            for (Map.Entry<String, Outer.MetacardAttribute> entry : metacardType.attributes.entrySet()) {
+                attributeDescriptors.add(attributeRegistry.getAttributeDescriptor(entry.getKey()));
+                if (entry.getValue().required) {
+                    requiredAttributes.add(entry.getKey());
+                }
+            }
+            if (!requiredAttributes.isEmpty()) {
+                ReportingMetacardValidator validator = new RequiredAttributesMetacardValidator(
+                        metacardType.type,
+                        requiredAttributes);
+                // TODO (RCZ) - do we need any service properties?
+                context.registerService(ReportingMetacardValidator.class, validator, null);
+
+            }
+            // TODO (RCZ) - What properties do we want?
+            Dictionary<String, ?> properties = new Hashtable<>();
+            MetacardType type = new MetacardTypeImpl(metacardType.type, attributeDescriptors);
+            context.registerService(MetacardType.class, type, properties);
+
+        }
+    }
+
+    private void parseValidators(Map<String, List<Outer.Validator>> validators,
+            String registrationId) {
+        for (Map.Entry<String, List<Outer.Validator>> entry : validators.entrySet()) {
+            List<AttributeValidator> attributeValidators = validatorFactory(entry.getValue());
+            attributeValidatorRegistry.registerValidators(entry.getKey(), attributeValidators);
+        }
+    }
+
+    private List<AttributeValidator> validatorFactory(List<Outer.Validator> validators) {
+        return validators.stream()
+                .filter(Objects::nonNull)
+                .filter(v -> StringUtils.isNotBlank(v.validator) && v.arguments != null)
```

Comment:
Remove `v.arguments != null` part of check (Arguments can be null, like pastdatevalidator)